src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -143,6 +143,24 @@ static String parseMapName(byte[] rawReplayBytes) {
     return mapPath.split("/")[2];
   }
 
+  @VisibleForTesting
+  static String parseCoopMapName(byte[] rawReplayBytes) {
+    // "ScenarioFile" in hex
+    byte[] startPattern = new byte[] {0x53, 0x63, 0x65, 0x6E, 0x61, 0x72, 0x69, 0x6F, 0x46, 0x69, 0x6C, 0x65, 0x00, 0x01};
+    int mapStartIndex = Bytes.indexOf(rawReplayBytes, startPattern) + startPattern.length;
+    int mapEndIndex = 0;
+
+    for (int i = mapStartIndex; i < rawReplayBytes.length - 1; i++) {
+      //0x00 0x01 is the field delimiter
+      if (rawReplayBytes[i] == 0x00 && rawReplayBytes[i+1] == 0x01)
+        break;
+      mapEndIndex = i;
+    }
+
+    String mapPath = new String(rawReplayBytes, mapStartIndex, mapEndIndex + 1 - mapStartIndex, US_ASCII);
+    return mapPath.split("/")[2];
```

Comment:
Is the map name always after the second slash?

Reply:
In reality it's not the map name but the map folder name. The path that I take from the binary data is the path to the script file.

So it looks like this :
/maps/my_awesome_map.v008/my_awesome_map.lua